app/script/entity/message/Message.js

```diff
@@ -0,0 +1,351 @@
+   * Check if message is a call message.
+   * @returns {Boolean} Is message of type call
+   */
+  is_call() {
+    return this.super_type === z.message.SuperType.CALL;
+  }
+
+  /**
+   * Check if message is a content message.
+   * @returns {Boolean} Is message of type content
+   */
+  is_content() {
+    return this.super_type === z.message.SuperType.CONTENT;
+  }
+
+  /**
+   * Check if message can be deleted.
+   * @returns {Boolean} True, if message is deletable.
+   */
+  is_deletable() {
+    if (this.is_ping() || !this.has_asset()) {
+      return true;
+    }
+    return ![z.assets.AssetTransferState.DOWNLOADING,
+      z.assets.AssetTransferState.UPLOADING].includes(this.get_first_asset().status());
+  }
+
+  /**
+   * Check if the message content can be downloaded
+   * @returns {Boolean} Is message of type content
+   */
+  is_downloadable() {
+    return ((typeof this.get_first_asset === 'function' ? this.get_first_asset().download : undefined) != null);
+  }
+
+  /**
+   * Check if message is a member message.
+   * @returns {Boolean} Is message of type member
+   */
+  is_member() {
+    return this.super_type === z.message.SuperType.MEMBER;
+  }
+
+  /**
+   * Check if message is a ping message.
+   * @returns {Boolean} Is message of type ping
+   */
+  is_ping() {
+    return this.super_type === z.message.SuperType.PING;
+  }
+
+  /**
+   * Check if message is a system message.
+   * @returns {Boolean} Is message of type system
+   */
+  is_system() {
+    return this.super_type === z.message.SuperType.SYSTEM;
+  }
+
+  /**
+   * Check if message is a e2ee message.
+   * @returns {Boolean} Is message of type system
+   */
+  is_device() {
+    return this.super_type === z.message.SuperType.DEVICE;
+  }
+
+  /**
+   * Check if message is a e2ee message.
+   * @returns {Boolean} Is message of type system
+   */
+  is_all_verified() {
+    return this.super_type === z.message.SuperType.ALL_VERIFIED;
+  }
+
+  /**
+   * Check if message is a e2ee message.
+   * @returns {Boolean} Is message of type system
+   */
+  is_unable_to_decrypt() {
+    return this.super_type === z.message.SuperType.UNABLE_TO_DECRYPT;
+  }
+
+  /**
+   * Check if message can be edited.
+   * @returns {Boolean} True, if message can be edited.
+   */
+  is_editable() {
+    return this.has_asset_text() && this.user().is_me && !this.is_ephemeral();
+  }
+
+  /**
+   * Check if message is ephemeral.
+   * @returns {Boolean} True, if message is ephemeral.
+   */
+  is_ephemeral() {
+    return this.ephemeral_expires() !== false;
+  }
+
+  /**
+   * Check if ephemeral message is expired.
+   * @returns {Boolean} True, if message expired.
+   */
+  is_expired() {
+    return this.ephemeral_expires() === true;
+  }
+
+  /**
+   * Check if message can be reacted to.
+   * @returns {Boolean} True, if message type supports reactions.
+   */
+  is_reactable() {
+    return this.is_content() && (this.status() !== z.message.StatusType.SENDING) && !this.is_ephemeral();
```

Comment:
`return this.is_content() && !this.is_ephemeral() && this.status() !== z.message.StatusType.SENDING;`